sound/soc/sof/pcm.c

```diff
@@ -24,7 +24,7 @@ static int create_page_table(struct snd_soc_component *component,
 			     unsigned char *dma_area, size_t size)
 {
 	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	struct snd_sof_dev *sdev = dev_get_drvdata(component->dev->parent);
```

Comment:
Sorry, but I have allergic reactions when we don't have a clear isolation between parent and child devices.
we shouldn't do this, this is a symptom that using sdev everywhere was a convenient solution that has exceeded its shelf life. Let's clean this up.

Reply:
@plbossart sure yes this is a bit ugly. I need to separate out the client (child) from the parent a bit more. will fix in the next version